src/traces/bar/plot.js

```diff
@@ -107,10 +107,12 @@ module.exports = function plot(gd, plotinfo, cdModule, traceLayer) {
             var prefix;
 
             if(trace.type === 'waterfall') {
-                var cont = trace[di.dir].marker;
-                lw = cont.line.width;
-                mc = cont.color;
-                prefix = 'waterfall';
+                if(!isBlank) {
+                    var cont = trace[di.dir].marker;
+                    lw = cont.line.width;
+                    mc = cont.color;
+                    prefix = 'waterfall';
```

Comment:
```
 prefix = 'waterfall';
```

is still well defined when `isBlank:true`, I'd leave it out of this block.

Reply:
Thanks for spotting that! Done in https://github.com/plotly/plotly.js/pull/3720/commits/be51cafe2d1d825ea1f3a3240a9f59c6c5cfa5ed.